app/javascript/articles/Feed.jsx

```diff
@@ -3,92 +3,263 @@ import { useEffect, useState } from 'preact/hooks';
+    }
 
-        // Remove that first story from the array to
-        // prevent it from rendering twice in the feed.
-        const featuredIndex = feedItems.indexOf(featuredStory);
-        if (featuredStory) {
-          feedItems.splice(featuredIndex, 1);
+    return filteredFeedPost;
+  }
+
+  // /**
+  //  * Inserts the billboards (if they exist) into the feed.
+  //  *
+  //  * @param {organizedFeedItems} The partially organized feed items.
+  //  * @param {String} feedFirstBillboard is the feed_first billboard retrieved from an endpoint.
+  //  * @param {String} feedSecondBillboard is the feed_second billboard retrieved from an endpoint.
+  //  * @param {String} feedThirdBillboard is the feed_third billboard retrieved from an endpoint.
+  //  *
+  //  * @returns {Array} We return the array containing the billboards slotted into the correct positions.
+  //  */
+  function insertBillboardsInFeed(
+    organizedFeedItems,
+    feedFirstBillboard,
+    feedSecondBillboard,
+    feedThirdBillboard,
+  ) {
+    if (organizedFeedItems.length >= 9 && feedThirdBillboard) {
+      organizedFeedItems.splice(7, 0, feedThirdBillboard);
+    }
+
+    if (organizedFeedItems.length >= 3 && feedSecondBillboard) {
+      organizedFeedItems.splice(2, 0, feedSecondBillboard);
+    }
+
+    if (organizedFeedItems.length >= 0 && feedFirstBillboard) {
+      organizedFeedItems.splice(0, 0, feedFirstBillboard);
+    }
+
+    return organizedFeedItems;
+  }
+
+  // /**
+  //  * Retrieves data for the feed. The data will include articles and billboards.
+  //  *
+  //  * @param {number} [page=1] Page of feed data to retrieve
+  //  * @param {string} The time frame of feed data to retrieve
+  //  *
+  //  * @returns {Promise} A promise containing the JSON response for the feed data.
+  //  */
+  async function fetchFeedItems(timeFrame = '', page = 1) {
+    const promises = [
+      fetch(`/stories/feed/${timeFrame}?page=${page}`, {
+        method: 'GET',
+        headers: {
+          Accept: 'application/json',
+          'X-CSRF-Token': window.csrfToken,
+          'Content-Type': 'application/json',
+        },
+        credentials: 'same-origin',
+      }),
+      fetch(`/display_ads/feed_first`),
+      fetch(`/display_ads/feed_second`),
+      fetch(`/display_ads/feed_third`),
+    ];
+
+    const results = await Promise.allSettled(promises);
+    const fullfilledPromises = [];
+    for (const result of results) {
+      if (result.status === 'fulfilled') {
+        let resolvedValue;
+        if (isJSON(result)) {
+          resolvedValue = await result.value.json();
         }
-        const organizedFeedItems = [featuredStory, feedItems].flat();
 
-        setFeedItems(organizedFeedItems);
-      } catch {
-        if (!onError) setOnError(true);
+        if (isHTML(result)) {
+          resolvedValue = await result.value.text();
+        }
+        fullfilledPromises.push(resolvedValue);
+      } else {
+        Honeybadger.notify(result.reason);
```

Comment:
@maestromac I wanted to double check with if this is the correct way (i.e. is this enough context) to notify honeybadger with or should I add any extra data?

Reply:
Hmm what does `result.reason` look like?

It's fine but doesn't hurt if you want to provide a little more context ie
```
    Honeybadger.notify("failed to fetch ...", context: { notification_id: notification.id })
```